src/ucp/wireup/wireup_cm.c

```diff
@@ -491,9 +493,13 @@ static unsigned ucp_cm_client_connect_progress(void *arg)
     if (context->config.ext.cm_use_all_devices) {
         /* Connect tmp_ep so it will send keepalive messages while the new lanes
          * are being connected */
-        ucp_ep_update_remote_id(wireup_ep->tmp_ep,
-                                progress_arg->sa_data->ep_id);
-        ucp_wireup_remote_connected(wireup_ep->tmp_ep);
+        ucp_ep_update_remote_id(tmp_ep, progress_arg->sa_data->ep_id);
+        for (lane = 0; lane < ucp_ep_num_lanes(tmp_ep); ++lane) {
+            if ((ucp_ep_config(tmp_ep)->key.cm_lane != lane) &&
+                ucp_wireup_ep_test(tmp_ep->uct_eps[lane])) {
```

Comment:
I thin they should be always wrapped by wireup_ep.
Do you see that the lane can be just UCT EP during client connect callback?
I'd change to assert that wireup_ep_test is always true